propose_sns.sh

```diff
@@ -60,5 +60,28 @@ sns propose \
     --test-neuron-proposer \
     "${SNS_CONFIGURATION_FILE_PATH}"
 
+# Save SNS canister IDs to sns_canister_ids.json
+NUM_SNS_INSTANCES=0
+while [ ${NUM_SNS_INSTANCES} -lt 1 ]
+do
+    dfx canister --network "${NETWORK}" \
+        call nns-sns-wasm list_deployed_snses '(record {})' \
+        | idl2json \
+        > sns-wasm-list_deployed_snses-response.json
+    NUM_SNS_INSTANCES=$(jq '.instances | length' sns-wasm-list_deployed_snses-response.json)
+    sleep 3
+done
+if [[ ${NUM_SNS_INSTANCES} -gt 1 ]]
+then
+    # TODO: pick the right SNS in case there are multiple ones present
+    echo "Error: sns-testing currently does not support multiple SNSes (found ${NUM_SNS_INSTANCES})"
```

Comment:
```suggestion
    echo "Error: sns-testing currently does not support multiple SNSes (found ${NUM_SNS_INSTANCES}). It is recommended to restart by using `cleanup.sh`"   
```

Or something along those lines